app/workers/github_issue_commenter.rb

```
@@ -0,0 +1,51 @@
+class GithubIssueCommenter
+  include Sidekiq::Worker
+  include Rails.application.routes.url_helpers
+  attr_reader :project, :branch, :pull_request, :github_service
+
+  def perform(project_id, branch_id, pull_request_data)
+    @pull_request = GithubPullRequest.new(pull_request_data)
+    @project = Project.find(project_id)
+    fetch_pull_request_number
+    return unless pull_request.number.present?
```

I would do something here to prevent a silent failure.